### ISSUE TITLE
Contribution Guide/Syntax Preferences

### ISSUE THREAD

**Issue by alexkrolick** [reactjs/rfcs#3]
Looking over the Context proposal (#2 ), it's using ~~~Stage-2~~~ _Stage-3_ ES and Flow typings, which are great, but for someone not familiar with Flow a bit hard to follow in parts. However, React's codebase _does_ use Flow, so it might be considered fair game. We've added syntax guidelines [to the docs]( https://github.com/reactjs/reactjs.org/blob/master/CONTRIBUTING.md#guidelines-for-code-examples) already, do you think something like that would be warranted for the RFCs, too?

**Comment by TrySound** [comment]
@alexkrolick Flow in the proposal is example of new api benefits for static type checking. Class fields are now in stage-3.

**Comment by acdlite** [comment]
Flow is helpful for documenting APIs, and proving that they can be soundly typed.

I'm not too worried about using stage 3 features in RFCs. We've avoided class property syntax in the docs because some of our audience may still be learning JavaScript. But I don't think that's a concern here. @gaearon may have opinions on this.

Syntax guidelines are a nice idea. A PR would be appreciated, if you're up for it.

**Comment by gaearon** [comment]
>@gaearon may have opinions on this.

My opinion is we actually *should* be using features we're pushing since that's the end game we're driving towards in RFCs.
